Dockerfile.wasi-builder

```diff
@@ -15,5 +17,9 @@ RUN wget https://github.com/WebAssembly/wasi-sdk/releases/download/${WASI_SDK}/$
       mkdir /wasi-sdk && \
       tar xf ${WASI_SDK}.0-linux.tar.gz --strip-components=1 -C ${WASI_SDK_ROOT} && \
       rm ${WASI_SDK}.0-linux.tar.gz
+RUN wget https://github.com/WebAssembly/binaryen/releases/download/version_${BINARYEN_VERSION}/binaryen-version_${BINARYEN_VERSION}-x86_64-linux.tar.gz && \
+      mkdir /binaryen && \
+      tar xf binaryen-version_${BINARYEN_VERSION}-x86_64-linux.tar.gz --strip-components=1 -C ${BINARYEN_ROOT} && \
+      rm binaryen-version_${BINARYEN_VERSION}-x86_64-linux.tar.gz
```

Comment:
This works for now, I would like to change this so that we have a previous image (that Dockerfile.wasi-builder builds FROM, as well as potentially other toolchains in the future) that contains binaryen and is able to make wasm-opt passes.

This would allow us to run this passes locally (as it's the goal of this change here), and we'll also be able to get rid of the [GH actions specific passes for that](https://github.com/vmware-labs/webassembly-language-runtimes/blob/d5e8a4a2b06320dfe45bed431ba4c43f482f6f6b/.github/workflows/release-php.yaml#L78-L108), prioritising local reproducibility.

I'm fine with this change as it is, we can refactor with my comments later.

Reply:
Implemented in https://github.com/vmware-labs/webassembly-language-runtimes/pull/33